### PR TITLE
Mlflow integration

### PR DESCRIPTION
**Pull request by pbobyrev**

### CODE REVIEW

File: src/syngen/ml/mlflow/mlflow_tracker.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        run_id: str = None,
+        experiment_id: Optional[str] = None,
+        run_name: Optional[str] = None,
+        nested: bool = False,
+        tags: Optional[Dict[str, Any]] = None,
+        description: Optional[str] = None
+    ):
+        print("START RUN IS RUNNNING")
+        if self.is_active:
+            print("IT IS ACTIVE")
```
[the remark this answers is not in the document]
Comment (pbobyrev) [reply src/syngen/ml/mlflow/mlflow_tracker.py:35]: I deleted it instead

File: src/syngen/ml/mlflow/mlflow_tracker.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    def start_run(
+        self,
+        run_id: str = None,
+        experiment_id: Optional[str] = None,
+        run_name: Optional[str] = None,
+        nested: bool = False,
+        tags: Optional[Dict[str, Any]] = None,
+        description: Optional[str] = None
+    ):
+        print("START RUN IS RUNNNING")
```
[the remark this answers is not in the document]
Comment (pbobyrev) [reply src/syngen/ml/mlflow/mlflow_tracker.py:33]: I deleted it instead

File: src/syngen/ml/metrics/accuracy_test/accuracy_test.py [hunk 3]
```diff
@@ -91,10 +92,24 @@ def report(self, **kwargs):
 
         uni_images = univariate.calculate_all(kwargs["cont_columns"], kwargs["categ_columns"], kwargs["date_columns"])
         bi_images = bivariate.calculate_all(kwargs["cont_columns"], kwargs["categ_columns"], kwargs["date_columns"])
-        correlations.calculate_all(kwargs["categ_columns"], kwargs["cont_columns"])
+        corr_result = correlations.calculate_all(kwargs["categ_columns"], kwargs["cont_columns"])
         clustering_result = "%.4f" % clustering.calculate_all(kwargs["categ_columns"], kwargs["cont_columns"])
         utility_result = utility.calculate_all(kwargs["categ_columns"], kwargs["cont_columns"])
 
+        tracker = MlflowTracker()
```
[the remark this answers is not in the document]
Comment (pbobyrev) [reply src/syngen/ml/metrics/accuracy_test/accuracy_test.py:99]: Is it feasible to make a one-line method?

File: src/syngen/infer.py [hunk 4]
```diff
@@ -81,8 +83,24 @@ def launch_infer(
         log_level=log_level,
         type="infer"
     )
+    mlflow_exp_name = f"exp at {datetime.today().strftime('%Y-%m-%d %H:%M:%S')}" if table_name else \
+        f"{metadata_path.split('/')[-1].split('.')[0]}_{datetime.today().strftime('%Y-%m-%d %H:%M:%S')}"
+    set_mlflow(mlflow_exp_name)
```
[the remark this answers is not in the document]
Comment (pbobyrev) [reply src/syngen/infer.py:88]: Do we have access to the cli-passed variables "metadata_path" and "table_name" outside of the "launch_infer()" method?

File: src/syngen/ml/mlflow/mlflow_tracker.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            self,
+            experiment_name: str = None,
+            experiment_id: str = None,
+    ):
+        """
+        Set the experiment for tracking.
+        If the experiment name is not provided, the last experiment will be used.
+        """
+        if self.is_active:
+            metadata_name = experiment_name[:-20]  # strip datetime
```
Comment (Anna050689) [inline src/syngen/ml/mlflow/mlflow_tracker.py:74]: This operation of indexing might cause an error.

File: src/syngen/ml/utils/utils.py [hunk 6]
```diff
@@ -258,25 +260,54 @@ def fetch_training_config(train_config_pickle_path):
         return pkl.load(f)
 
 
-def create_log_file(type_of_process: str, table_name: str, metadata_path: str):
-    """
-    Create the file for storing the logs of main processes
-    """
+def fetch_unique_root(table_name: str, metadata_path: str):
```
Comment (Anna050689) [inline src/syngen/ml/utils/utils.py:263]: Please, add the docstring here

File: src/syngen/ml/utils/utils.py [hunk 7]
```diff
@@ -258,25 +260,54 @@ def fetch_training_config(train_config_pickle_path):
...
 
+def set_mlflow_exp_name(table_name: str, metadata_path: str):
+    name = fetch_unique_root(table_name, metadata_path)
+    os.environ["MLFLOW_EXPERIMENT_NAME"] = name
+
+
+def set_mlflow(type_of_process: str):
+    exp_name = os.environ["MLFLOW_EXPERIMENT_NAME"]
+    try:
+        response = os.system("ping -c 1 " + os.environ.get("MLFLOW_TRACKING_URI")[7:-6])
```
Comment (Anna050689) [inline src/syngen/ml/utils/utils.py:294]: Is it possible to reduce the process of indexing "ML_FLOW_TRACKING_URL"?

**Review by Anna050689 — CHANGES_REQUESTED** [review]

**Review by Anna050689 — APPROVED** [review]

**Pull request merged**